File: jiwon/programmers/c++/secret_map.cpp
#include <string>
#include <iostream>
#include <vector>
#include <deque>

using namespace std;


vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    for (int i = 0; i < arr1.size(); i++) {
        int num = arr1[i] | arr2[i];
        deque<string> q(n);
        string str = "";
        while (num > 0) {
            if (num % 2 == 1) q.push_back("#");
            else q.push_back(" ");
            num /= 2;
        }
        while (q.size() < n * 2) q.push_back(" ");
        while (q.size() > 0) {
            str.append(q.back());
            q.pop_back();
        }
        answer.push_back(str);

    }
    return answer;
}

File: jiwon/programmers/python/Secret_Map.py
def get_wall_board(n, arr):
    board = []
    for item in arr:
        num = 2 ** n
        row = ""
        for _ in range(n, -1, -1):
            if item >= num:
                item -= num
                row += "#"
            else:
                row += " "
            num //= 2
        board.append(row)
    return board


def solution(n, arr1, arr2):
    answer = []
    board1 = get_wall_board(n - 1, arr1)
    board2 = get_wall_board(n - 1, arr2)
    for i in range(n):
        row = ""
        for j in range(n):
            if board1[i][j] == "#" or board2[i][j] == "#":
                row += "#"
            else:
                row += " "
        answer.append(row)
    return answer